test/jdk/java/nio/channels/spi/SelectorProvider/inheritedChannel/UnixDomainChannelTest.java

```diff
@@ -64,54 +73,92 @@ public static void main(String args[]) throws Exception {
             throw new RuntimeException();
     }
 
-    private static void closeAll(UnixDomainSocket... sockets) {
-        for (UnixDomainSocket sock : sockets) {
-            sock.close();
-        }
-    }
-
     // Test with a named connected socket
     private static void test1() throws Exception {
-        UnixDomainSocket listener = new UnixDomainSocket();
-        listener.bind("foo.socket");
-        UnixDomainSocket sock1 = new UnixDomainSocket();
-        sock1.connect("foo.socket");
-        UnixDomainSocket sock2 = listener.accept();
-
-        Launcher.launchWithUnixDomainSocket("UnixDomainChannelTest$Child", sock2, "test1");
-        int c = sock1.read();
-        if (c != 'Y') {
-            System.err.printf("test1: failed %d d\n", c );
+        ServerSocketChannel listener = ServerSocketChannel.open(UNIX);
```

Comment:
Maybe try with resources or try-finally could be used to ensure that all channels are closed.